lib/src/client/http2_connection.dart

```diff
@@ -77,19 +77,13 @@ class Http2ClientConnection implements connection.ClientConnection {
 
   Future<ClientTransportConnection> connectTransport() async {
     final securityContext = credentials.securityContext;
-    Socket socket;
-    if (securityContext == null) {
-      socket = await Socket.connect(host, port);
-    } else {
-      socket = await SecureSocket.connect(host, port,
-          supportedProtocols: ['h2'],
+    Socket socket = await Socket.connect(host, port);
+    if (securityContext != null) {
+      // Todo(sigurdm): We want to pass supportedProtocols: ['h2']. http://dartbug.com/37950
+      socket = await SecureSocket.secure(socket,
+          host: authority,
```

Comment:
If `host == authority` as is the common case, you could still use `connect`. But this seems like an over optimization :)

Reply:
Yes - I thought it would be even more cases for our undertested framework

I hope we solve http://dartbug.com/37950 and move to a better world.